### ISSUE TITLE
ContextMenu data binding support and behaviour is vastly different than Menu

### ISSUE THREAD

**Issue by MarchingCube** [AvaloniaUI/Avalonia#2325]
So I tried to bind to `Items` property of a `ContextMenu` in a same way control catalog handles `Menu`. But it seems that it works differently, and I only get the view model class name, and no style properties seem to  be applied.
![image](https://user-images.githubusercontent.com/2588062/53245496-29806300-36ae-11e9-961f-bfe736a88290.png)

I did that in my branch by simply copying relevant bits from `MenuPage` over to `ContextMenuPage`. 
https://github.com/MarchingCube/Avalonia/commits/feature/context-menu

My attempt:
https://github.com/MarchingCube/Avalonia/blob/a09446520c824b108c5f67c2e719fb75b87c84a1/samples/ControlCatalog/Pages/ContextMenuPage.xaml#L38-L60

**Comment by Gillibald** [comment]
Looks like ContextMenu has not the required ItemGenerator bits that's why you don't see menu items that wrap bound viewmodel.

**Comment by MarchingCube** [comment]
@grokys @Gillibald So I decided to check if adding the container generator will fix it. Just copying container generator fixed this issue, but there was other other issue that popped up (styles got messed up for separators and sub menus were not closing properly), The issue with items can be fixed by copying even more bits from `Menu` like `SubmenuOpenedEvent` handling.

But at this point I just wonder why there is no base class for `Menu` and `ContextMenu` as both classes have quite some code in common (and probably will have even more after unifying behavior) and there is quite some copy paste action going on there.

Additionally the keyboard behaviour is different, in WPF when you open context menu user can immediately navigate the context menu using keyboard. This is not the case for the `ContextMenu`, but it works for `Menu`.

All issues cause the `ContextMenu` to be not that useful outside of hardcoded xaml scenarios. And probably confusing to the users, as people will probably expect correct keyboard behaviour.

To sum up my points:
- we probably need a base class for common menu stuff
- styles most likely can be shared as well
- keyboard behaviour should be consistent

**Comment by kekekeks** [comment]
While you are here unifying menu behaviors. See https://github.com/AvaloniaUI/Avalonia/issues/898#issuecomment-443598096

If we want somewhat consistent behavior we also need a way to databind exported menus. Which is a problem because everything in item generation expects a control. It's even an Item*Container*Generator. We need some way to convert lists from the view model to non-control lists of children.

**Comment by MarchingCube** [comment]
@kekekeks I don't have much UI experience outside of Windows so some of the concepts in that issue are new to me. So it would be nice to figure out some plan on what/how/when can be fixed. Like from my standpoint in one of my applications I already have issues with `ContextMenu` misbehaving and generally having quirky behaviour. Right now I only run on Windows, but obviously I might need to investigate other platforms this year.

**Comment by MarchingCube** [comment]
Part of this issue seem to be reported here: https://github.com/AvaloniaUI/Avalonia/issues/1466
